### PR TITLE
Fix dylib references of bundled programs on macOS

### LINKED ISSUE #2134: bech32, cardano-{address,tx} release executables don't work on macOS without nix
# Context

https://input-output-rnd.slack.com/archives/GBT05825V/p1599837130089800?thread_ts=1599834710.084500&cid=GBT05825V
 
 

| Information  | -                                                     |
| ---          | ---                                                   |
| Version      | multiple ? |
| Platform     | macOS     |
| Installation | Github Release            |

# Steps to Reproduce

1. Download and extract a release archive
2. `otool -L ./cardano-address` (and `cardano-addres`, `bech32`)

## Expected behavior

The dylib paths are relative to `@executable_path`, like for cardano-wallet

```bash
./cardano-wallet:
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	@executable_path/libz.dylib (compatibility version 1.0.0, current version 1.2.11)
	@executable_path/libsodium.23.dylib (compatibility version 27.0.0, current version 27.0.0)
	@executable_path/libiconv.dylib (compatibility version 7.0.0, current version 7.0.0)
	@executable_path/libgmp.10.dylib (compatibility version 15.0.0, current version 15.0.0)
	@executable_path/libffi.7.dylib (compatibility version 9.0.0, current version 9.0.0)
```

## Actual behavior

```bash
otool -L ./cardano-address
./cardano-address:
	/nix/store/f573hx6rga70hx5xjqrvqa58jkdxdl91-Libsystem-osx-10.12.6/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1226.10.1)
	/nix/store/p95rm5rlv4fmi0k81nr0l29ax57dw1r0-zlib-1.2.11/lib/libz.dylib (compatibility version 1.0.0, current version 1.2.11)
	/nix/store/m3yqw7xcmhvd0cmm3jh3b469kxf3yhfp-libiconv-osx-10.12.6/lib/libiconv.dylib (compatibility version 7.0.0, current version 7.0.0)
	/nix/store/hrmqighslwzk32rmz00k1lw5k53f5jr3-gmp-6.2.0/lib/libgmp.10.dylib (compatibility version 15.0.0, current version 15.0.0)
	/nix/store/w1v69v7w418nvsc3rxndp4cq59110bzj-libffi-3.3/lib/libffi.7.dylib (compatibility version 9.0.0, current version 9.0.0)
```

Reported by @piotr-iohk to cause the binaries to be un-runnable on a macOS without nix installed (and you'd need the .dylibs to actually be in the store somehow, not _just_ to have nix installed)
 

---

# Resolution 

---

# QA

### PR DESCRIPTION
**Pull request by rvl**
### Issue Number

Resolves #2134

### Overview

- On macOS rewrite dylib references from `/nix/store` to `@executable_path` - for every executable in the bundle.
- Add linking tests to `check-bundle.rb` for macOS and Linux.

### CODE REVIEW

File: scripts/check-bundle.rb [hunk 1]
```diff
@@ -33,15 +35,28 @@ def report(cmd, status)
...
-report(cmd, status.to_i)
+begin
+  ver, status = Open3.capture2("#{runner} #{cmd}")
+  report(cmd, status.to_i)
+rescue Errno::ENOENT
+  report(cmd, 1)
+end
 
-tests[wallet].each do |cmd|
+tests[wallet].unshift(wallet).each do |cmd|
```
Comment (piotr-iohk) [inline scripts/check-bundle.rb:45]: :)

**Review by piotr-iohk — APPROVED** [review]
:+1:

File: scripts/check-bundle.rb (hunk 1 above)
Comment (rvl) [reply scripts/check-bundle.rb:45]: My first ruby program :slightly_smiling_face:

**Pull request merged**